lib/charms/layer/vault_kv.py

```diff
@@ -144,10 +145,12 @@ class VaultAppKV(_VaultBaseKV):
 
     def __init__(self, *_, **kwds):
         self._kwds = kwds
-        self._path = "{}/kv/app".format(self._config["secret_backend"])
-        self._hash_path = "{}/kv/app-hashes/{}".format(
-            self._config["secret_backend"], hookenv.local_unit().split("/")[1]
-        )
+        # self._kwds attrribute must be set first
+        # as _config attribute is based off its values
+        backend = self._config["secret_backend"]
+        app = hookenv.local_unit().split("/")[1]
+        self._path = f"{backend}/kv/app"
+        self._hash_path = f"{backend}/kv/app-hashes/{app}"
```

Comment:
If `hookenv.local_unit()` is `kubernetes-control-plane/0`, then `hookenv.local_unit().split("/")[1]` would be `0`. I don't think `app` is quite the right term for it. Can we call it a unit number instead?

```suggestion
        unit_num = hookenv.local_unit().split("/")[1]
        self._path = f"{backend}/kv/app"
        self._hash_path = f"{backend}/kv/app-hashes/{unit_num}"
```